app/controllers/staff/speakers_controller.rb

```diff
@@ -1,36 +1,26 @@
 class Staff::SpeakersController < Staff::ApplicationController
   decorates_assigned :speaker
-  before_filter :set_proposal, only: [:new, :create, :destroy]
+  before_filter :set_program_session, only: [:new, :create, :destroy]
 
   def index
-    render locals: {
-             proposals: @event.proposals.includes(speakers: :user).decorate
-           }
+    @program_session_speakers = Speaker.with_program_session.for_event(current_event)
   end
 
   def new
     @speaker = Speaker.new
-    @user = @speaker.build_user
-    @proposal = Proposal.find_by(uuid: params[:proposal_uuid])
   end
 
   def create
     s_params = speaker_params
-    user = User.find_by(email: s_params.delete(:speaker_email))
-    if user
-      if user.speakers.create(s_params.merge(proposal: @proposal))
-        flash[:success] = "Speaker was added to this proposal"
-      else
-        flash[:danger] = "There was a problem saving this speaker"
-      end
+    @speaker = @program_session.speakers.create(s_params.merge(program_session: @program_session, event: current_event))
+    if @speaker.save
+      flash[:success] = "#{@speaker.name} has been added to #{@program_session.title}"
     else
-      flash[:danger] = "Could not find a user with this email address"
+      flash[:danger] = "There was a problem saving this speaker."
     end
-    redirect_to event_staff_proposal_url(event, @proposal)
+    redirect_to event_staff_speakers_path(current_event)
   end
 
-  #if user input (params), exist
-
   def show
     @speaker = Speaker.find(params[:id])
```

Comment:
All Speaker.find type calls should be scoped by current_event so you can't accidentally find a speaker from another event.